com.unity.netcode.gameobjects/Runtime/Core/SnapshotSystem.cs

```diff
@@ -508,6 +682,43 @@ internal void HandleSnapshot(ulong clientId, SnapshotDataMessage message)
                 SpawnObject(spawnCommand, clientId);
             }
 
+            for (int index = 0; index < header.UpdateCount; index++)
+            {
+                byte[] readBuffer = new byte[10000];
+                // todo: this large buffer is very inelegant. Add proper bounds check to the unsafe code below
+                // and also move into a member reader
+
+                fixed (byte* buff = &readBuffer[0])
+                {
+                    message.ReadBuffer.TryBeginRead(FastBufferWriter.GetWriteSize(updateCommand));
+                    message.ReadBuffer.ReadValue(out updateCommand);
+
+                    message.ReadBuffer.TryBeginRead(updateCommand.SerializedLength);
+                    message.ReadBuffer.ReadBytes(buff, updateCommand.SerializedLength, 0);
+
+                    //NetworkVariableBase variable;
+                    GetBehaviourVariable(updateCommand, out NetworkBehaviour behaviour, out NetworkVariableBase variable, clientId);
+
+                    if (updateCommand.TickWritten > variable.TickRead)
+                    {
+                        using var reader =
+                            new FastBufferReader(buff, Allocator.Temp, updateCommand.SerializedLength, 0);
```

Comment:
nit
```suggestion
                        using var reader = new FastBufferReader(buff, Allocator.Temp, updateCommand.SerializedLength, 0);
```

Reply:
Yeah. But wait for next commit :-) The whole `new FastBufferReader` is gone. I'll pass the `message.ReadBuffer` in instead, and just skip over if the variable is to not be deserialized. Less allocation, less copy. win-win.